featurebyte/service/feature_list.py

```diff
@@ -169,23 +168,20 @@ async def _extract_relationships_info(
 
     async def _update_features(
         self,
-        features: list[FeatureModel],
+        feature_ids: Sequence[ObjectId],
         inserted_feature_list_id: Optional[ObjectId] = None,
         deleted_feature_list_id: Optional[ObjectId] = None,
     ) -> None:
-        for feature in features:
-            feature_list_ids = cast(List[ObjectId], feature.feature_list_ids)
-            if inserted_feature_list_id:
-                feature_list_ids = self.include_object_id(
-                    feature_list_ids, inserted_feature_list_id
-                )
-            if deleted_feature_list_id:
-                feature_list_ids = self.exclude_object_id(feature_list_ids, deleted_feature_list_id)
-            await self.feature_service.update_document(
-                document_id=feature.id,
-                data=FeatureServiceUpdate(feature_list_ids=feature_list_ids),
-                document=feature,
-                return_document=False,
+        if inserted_feature_list_id:
+            await self.feature_service.update_documents(
+                query_filter={"_id": {"$in": feature_ids}},
+                update={"$addToSet": {"feature_list_ids": inserted_feature_list_id}},
+            )
+
+        if deleted_feature_list_id:
+            await self.feature_service.update_documents(
+                query_filter={"_id": {"$in": feature_ids}},
+                update={"$pull": {"feature_list_ids": deleted_feature_list_id}},
             )
```

Comment:
Based on the feature list of 204 features on my local machine, this parts reduces the mongo persistent update time from 0.8s to 0.1s.